src/patient_representation/tl/basic.py

```diff
@@ -1641,3 +1703,241 @@ def calculate_distance_matrix(self, force: bool = False):
+            sample_size_threshold=sample_size_threshold,
+            cluster_size_threshold=cluster_size_threshold,
+        )
+
+        if self.aggregate_cell_types:
+            # Aggregate by BOTH sample and cell type
+            pseudobulk_data = self._get_pseudobulk(
+                aggregation=self.aggregation_mode, fill_value=np.nan, aggregate_cell_types=True
+            )
+            self.views = [[view_matrix] for view_matrix in pseudobulk_data]  # -> multiple  celltype view appraoch
+            self.views_names = self.cell_types
+        else:
+            # Aggregate ONLY by patient
+            pseudobulk_data = self._get_pseudobulk(
+                aggregation=self.aggregation_mode, fill_value=np.nan, aggregate_cell_types=False
+            )
+            self.views = [[pseudobulk_data]]  # -> single view appraoch
+            self.views_names = ["aggregated_gene_expression"]
+
+        ent = entry_point()
+
+        ent.set_data_options(**self.data_options)
+
+        ent.set_data_matrix(
+            data=self.views,
+            samples_names=[self.samples],
+            views_names=self.views_names,
+            groups_names=[
+                "group1"
+            ],  # All patients are considered as a single group; no group-specific modeling is needed
+        )
+
+        ent.set_model_options(**self.model_options)
+
+        ent.set_train_options(**self.train_options)
+
+        ent.build()
+        ent.run()
+
+        self.model = ent.model
+
+    def calculate_distance_matrix(self, force=False, store_weights=False, dist="euclidean"):
+        """
+        Calculate distances between patients using MOFA2 latent factors.
+
+        Parameters
+        ----------
+        force : bool = False
+            If True, recalculate the distance matrix even if it exists.
+        store_weights : bool, default: False
+            If True, store the weights (relation of factors to genes) in `self.adata.uns`.
+
+        Returns
+        -------
+        distances : np.ndarray
+            Matrix of distances between patients.
+        """
+        distances = super().calculate_distance_matrix(force=force)
+        if distances is not None:
+            return distances
+
+        distance_metric = valid_distance_metric(dist)
+
+        # get factors expectation (latent representations of samples)
+        self.patient_representation = self.model.nodes["Z"].getExpectation()  # Shape: (n_patients, n_factors)
+
+        # store weights (relation of factors to genes)
+        if store_weights:
+            weights = self.model.nodes["W"].getExpectation()
+            # weights is a list with one matrix per view
+            if self.aggregate_cell_types:
+                mofa_weights = {view_name: weights[i] for i, view_name in enumerate(self.views_names)}
+            else:
+                mofa_weights = weights[0]
+
+        distances = scipy.spatial.distance.pdist(self.patient_representation, metric=distance_metric)
```

Comment:
These 2 lines are repeated in several classes, might be worth moving to a new method `_representation_to_distances()`